### PR TITLE
fix(chips): do not set chips value if there's no ngControl or value

### LINKED ISSUE #7261: Chip-list overrides initial chip selected state
Say you have a template like this:
```html
<mat-chip-list>
  <mat-chip *ngFor="let i of items" selected="true"> {{i}} </mat-chip>
</mat-chip-list>
```

Because of this async behavior in chip-list:
https://github.com/angular/material2/blob/3571f68a2de0833ee4741af8a0eb24d1da174f38/src/lib/chips/chip-list.ts#L574-L577

The individual chip selection gets overwritten by the chip-list on init.

@tinayuangao we should find a way to avoid using `Promise.resolve` here

### PR DESCRIPTION
**Pull request by tinayuangao**
Fixes #7261

### CODE REVIEW

File: src/lib/chips/chip-list.ts [hunk 1]
```diff
@@ -572,8 +572,10 @@ export class MdChipList implements MdFormFieldControl<any>, ControlValueAccessor
     // Defer setting the value in order to avoid the "Expression
     // has changed after it was checked" errors from Angular.
     Promise.resolve().then(() => {
-      this._setSelectionByValue(this.ngControl ? this.ngControl.value : this._value, false);
-      this.stateChanges.next();
+      if (!!this.ngControl || !!this._value) {
```
Comment (jelbourn) [inline src/lib/chips/chip-list.ts]: You don't need the `!!` in conditional expressions, only when the value will be passed around somewhere

Comment (tinayuangao) [reply src/lib/chips/chip-list.ts]: Done. PTAL. Thanks for review.

Comment (jelbourn) [reply src/lib/chips/chip-list.ts]: Need to push your changes?

Comment (tinayuangao) [reply src/lib/chips/chip-list.ts]: Changes are pushed. Thanks!

**Pull request merged**